tests/Agent/IntegrationTests/SharedApplications/Common/SharedApplicationHelpers/ReflectionHelpers/ReflectionUtil.cs

```diff
@@ -79,7 +79,18 @@ public static List<Type> GetClassesByType<T>(Assembly assembly)
 
             foreach (var assembly in assemblies)
             {
-                foreach (var type in assembly.GetTypes())
+                Type[] types;
+                try
+                {
+                    types = assembly.GetTypes();
+                }
+                catch (Exception)
+                {
+                    Console.WriteLine($"Exception calling .GetTypes on assembly {assembly.FullName}");
+                    throw;
```

Comment:
Don't we want to continue if we hit a bad assembly? Or do you want it to fail out?

Reply:
I'd rather have it die, as it's a sign of a bigger problem.